File: Mathematical Algorithms/Prime Triplet/PTriplet.java
import java.util.Arrays;

public class PTriplet {
    static void sieve(int n, boolean prime[]) {
        for (int p = 2; p * p <= n; p++) {
            if (prime[p] == true) {
                for (int i = p * 2; i <= n; i += p)
                    prime[i] = false;
            }
        }
    }

    // function to print prime triplets
    static void printPrimeTriplets(int n) {
        boolean prime[] = new boolean[n + 1];
        Arrays.fill(prime, true);
        sieve(n, prime);
        System.out.println("The prime triplets" + " from 1 to " + n + "are :");
        for (int i = 2; i <= n - 6; ++i) {
            // triplets of form (p, p+2, p+6)
            if (prime[i] && prime[i + 2] && prime[i + 6])
                System.out.println(i + " " + (i + 2) + " " + (i + 6));
                // triplets of form (p, p+4, p+6)
            else if (prime[i] && prime[i + 4] &&
                    prime[i + 6])
                System.out.println(i + " " + (i + 4) + " " + (i + 6));
        }
    }
}

File: Mathematical Algorithms/Prime Triplet/PTriplet.py
def sieve(n, prime):
    p = 2
    while p * p <= n:
        if prime[p] == True:
            i = p * 2
            while i <= n:
                prime[i] = False
                i = i + p
        p = p + 1


# function to print prime triplets
def printPrimeTriplets(n):
    prime = [True] * (n + 1)
    sieve(n, prime)
    print("The prime triplets from 1 to ", n, "are :")
    for i in range(2, n - 6 + 1):
        # triplets of form (p, p+2, p+6)
        if (prime[i] and prime[i + 2] and
                prime[i + 6]):
            print(i, (i + 2), (i + 6))
        # triplets of form (p, p+4, p+6)
        elif (prime[i] and prime[i + 4] and
              prime[i + 6]):
            print(i, (i + 4), (i + 6))
